### PR TITLE
FIX: Err handling

### PR DESCRIPTION
**Pull request by cshezi**
Error handling, fixed prettier and
changed how to get rule cfg early exit

### CODE REVIEW

**Comment by johanfol** [comment]
Most changes on here is in another open PR - only change on determineOutcome function in here doesn't exist on the other branch - so either do the change in that branch, or wait for that merge to come through, then merge main back into your branch to sort out conflicts. Best suggestion is apply the determineOutcome changes to the other open branch. 
https://github.com/frmscoe/rule-executer/pull/4

File: src/controllers/execute.ts [hunk 1]
```diff
@@ -41,34 +42,46 @@ export const execute = async (ctx: Context): Promise<void | Context> => {
...
+  if (!ruleConfig) {
+    ctx.body = {
+      ruleResult: {
+        ...ruleRes,
+        reason: 'Rule processor configuration not retrievable',
+      },
+      transaction: request.transaction,
+      networkSubMap: request.networkMap,
+    };
+    return ctx;
```
Comment (johanfol) [inline src/controllers/execute.ts:81]: add ctx.status = 500

File: src/helpers/determineOutcome.ts [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+import { loggerService } from '..';
 
 const determineOutcome = (
   value: number,
   ruleConfig: RuleConfig,
   ruleResult: RuleResult,
 ): RuleResult => {
-  if (value || value === 0) {
-    for (const band of ruleConfig.config.bands) {
+  if (ruleConfig.config.bands && ruleConfig.config.case) {
```
Comment (johanfol) [inline src/helpers/determineOutcome.ts:9]: why should both bands && case be configured? Shouldn't this be || ?

**Review by johanfol — CHANGES_REQUESTED** [review]

File: src/helpers/determineOutcome.ts (hunk 2 above)
Comment (cshezi) [reply src/helpers/determineOutcome.ts:9]: On this condition we are checking if both exist in the configure which I believe is correct logic to handle this case

Comment (johanfol) [reply src/helpers/determineOutcome.ts:9]: but both won't exist for rules. config will always either be bands or case, never both

File: src/controllers/execute.ts [hunk 3]
```diff
@@ -41,34 +42,47 @@ export const execute = async (ctx: Context): Promise<void | Context> => {
...
+  if (!ruleConfig) {
+    ctx.body = {
+      ruleResult: {
+        ...ruleRes,
+        reason: 'Rule processor configuration not retrievable',
+      },
+      transaction: request.transaction,
+      networkSubMap: request.networkMap,
+    };
+    ctx.status = 500;
```
Comment (cshezi) [inline src/controllers/execute.ts:81]: Changes have been make

**Review by johanfol — APPROVED** [review]

**Pull request merged**